tracer/src/Datadog.Trace/DuckTyping/IDuckTypeTask.cs

```diff
@@ -0,0 +1,47 @@
+// <copyright file="IDuckTypeTask.cs" company="Datadog">
+// Unless explicitly stated otherwise all files in this repository are licensed under the Apache 2 License.
+// This product includes software developed at Datadog (https://www.datadoghq.com/). Copyright 2017 Datadog, Inc.
+// </copyright>
+
+#nullable enable
+namespace Datadog.Trace.DuckTyping;
+
+/// <summary>
+/// Duck type task interface
+/// </summary>
+/// <typeparam name="T">Type of the result</typeparam>
+public interface IDuckTypeTask<out T> : IDuckType
```

Comment:
This seems like it is pretty much a direct overlap with https://github.com/DataDog/dd-trace-dotnet/pull/6480... I feel like we should maybe try to unify those with these at least? 🤔 Not in this PR, just a thought.

Reply:
Yes we need to do that in a different PR